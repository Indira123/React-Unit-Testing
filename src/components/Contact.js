import React from "react";

const Contact = (props) => {
  return <span id={props.phone}>{props.phone}</span>;
};

export default Contact;
