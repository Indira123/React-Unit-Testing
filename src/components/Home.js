import React, { useState } from 'react';
import Contact from './Contact';

function Home() {
    const [title, setTitle] = useState('');

    const handleChange = (event) => {
        setTitle(event.target.value)
    }
    const handleSubmit = (event) => {
        event.preventDefault();
    }
    return (
        <div id='myAppElement'>
            <form onSubmit={handleSubmit} style={{ width: 400 + 'px' }}>
                <div className='form-group'>
                    <label htmlFor="title">Item</label>
                    <input
                        type="text"
                        id="title"
                        value={title}
                        onChange={handleChange}
                        className="form-control"
                    />
                </div>
                <button className='btn btn-primary' type="submit">SAVE</button>
            </form>
            <Contact phone={'9999999999'} />
        </div>
    );
}

export default Home;