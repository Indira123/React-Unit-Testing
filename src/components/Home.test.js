import React from 'react';
import { shallow, mount } from 'enzyme';
import Home from './Home';
import Contact from './Contact';

it('Home component snapshot', () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
});
test('Home component exists', () => {
    const component = shallow(<Home />);
    expect(component.exists()).toBe(true)
});
test('Home Prevent Submit', () => {
    const component = shallow(<Home />);
    let prevented = false;
    component.find('form').simulate('submit', {
        preventDefault: () => {
            prevented = true;
        }
    })
    expect(prevented).toBe(true)
})
//integration testing
test('render contact when phone changes', () => {
    const component = mount(<Contact phone='' />);
    component.setProps({ phone: '77777777' });
    expect(component.find('span').prop('id')).toEqual('77777777')
});

