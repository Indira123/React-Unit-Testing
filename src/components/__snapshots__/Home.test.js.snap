// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `ShallowWrapper {}`;

exports[`Home should render correctly 1`] = `ShallowWrapper {}`;

exports[`Home should render correctly 2`] = `ShallowWrapper {}`;
